databricks/koalas/frame.py

```diff
@@ -413,7 +413,73 @@ def __init__(self, data=None, index=None, columns=None, dtype=None, copy=False):
                 pdf = data
             else:
                 pdf = pd.DataFrame(data=data, index=index, columns=columns, dtype=dtype, copy=copy)
-            super(DataFrame, self).__init__(InternalFrame.from_pandas(pdf))
+            internal = InternalFrame.from_pandas(pdf)
+
+        self._internal_frame = internal
+
+    @property
+    def _ksers(self):
+        from databricks.koalas.series import Series
+
+        if not hasattr(self, "_kseries"):
+            self._kseries = {
+                label: Series(data=self, index=label) for label in self._internal.column_labels
+            }
+        else:
+            kseries = self._kseries
+            assert len(self._internal.column_labels) == len(kseries), (
+                len(self._internal.column_labels),
+                len(kseries),
+            )
+            if any(
+                self is not kser._kdf or not same_anchor(self, kser) for kser in kseries.values()
+            ):
+                self._kseries = {
+                    label: kseries[label]
+                    if self is kseries[label]._kdf and same_anchor(self, kseries[label])
+                    else Series(data=self, index=label)
+                    for label in self._internal.column_labels
+                }
+        return self._kseries
+
+    @property
+    def _internal(self) -> InternalFrame:
+        return self._internal_frame
+
+    def _update_internal_frame(self, internal: InternalFrame, requires_same_anchor: bool = True):
+        from databricks.koalas.series import Series
+
+        kseries = {}
+
+        for old_label, new_label in zip_longest(
+            self._internal.column_labels, internal.column_labels
+        ):
+            if old_label is not None:
+                kser = self._ksers[old_label]
+                if old_label != new_label or (
```

Comment:
Do we assume the anchor is different 1. if a position of a label was changed 2. if the label was changed (?). Might be best to add comments here ..